### ISSUE TITLE
NetMQBeacon throws on Linux with Mono

### ISSUE THREAD

**Issue by drewnoakes** [zeromq/netmq#249]
Running unit tests on Ubuntu 14.04 via Mono shows:

``` none
NUnit-Console version 2.6.2.12296
Copyright (C) 2002-2012 Charlie Poole.
Copyright (C) 2002-2004 James W. Newkirk, Michael C. Two, Alexei A. Vorontsov.
Copyright (C) 2000-2002 Philip Craig.
All Rights Reserved.

Runtime Environment - 
   OS Version: Unix 3.13.0.45
  CLR Version: 4.0.30319.17020 ( Mono 4.0 ( 3.12.0 (tarball Sat Feb  7 19:12:57 UTC 2015) ) )

ProcessModel: Default    DomainUsage: Single
Execution Runtime: mono-4.0
..[ERROR] FATAL UNHANDLED EXCEPTION: System.NotImplementedException: The requested feature is not implemented.
  at System.Net.NetworkInformation.LinuxUnicastIPAddressInformation.get_IPv4Mask () [0x00000] in <filename unknown>:0 
  at NetMQ.InterfaceCollection..ctor () [0x00081] in /home/drew/dev/netmq/src/NetMQ/InterfaceCollection.cs:43 
  at NetMQ.NetMQBeacon+Shim.Configure (System.String interfaceName, Int32 port) [0x000a3] in /home/drew/dev/netmq/src/NetMQ/NetMQBeacon.cs:77 
  at NetMQ.NetMQBeacon+Shim.OnPipeReady (System.Object sender, NetMQ.NetMQSocketEventArgs e) [0x000a0] in /home/drew/dev/netmq/src/NetMQ/NetMQBeacon.cs:201 
  at NetMQ.NetMQSocket.InvokeEvents (System.Object sender, PollEvents events) [0x00048] in /home/drew/dev/netmq/src/NetMQ/NetMQSocket.cs:242 
  at NetMQ.Poller.PollWhile (System.Func`1 condition) [0x00253] in /home/drew/dev/netmq/src/NetMQ/Poller.cs:408 
  at NetMQ.Poller.PollTillCancelled () [0x0000e] in /home/drew/dev/netmq/src/NetMQ/Poller.cs:296 
  at NetMQ.NetMQBeacon+Shim.Run (NetMQ.Sockets.PairSocket shim) [0x00087] in /home/drew/dev/netmq/src/NetMQ/NetMQBeacon.cs:145 
  at NetMQ.NetMQActor.RunShim () [0x0000e] in /home/drew/dev/netmq/src/NetMQ/NetMQActor.cs:150 
  at System.Threading.Thread.StartInternal () [0x00000] in <filename unknown>:0 
```

`UnicastAddress.IPv4Mask` throws `NotImplementedException`. The underlying problem is an incomplete implementation of `LinuxUnicastIPAddressInformation` in Mono.

This bug has been known since Nov 2011: https://bugzilla.xamarin.com/show_bug.cgi?id=2033

Workarounds exist, but use `[DllImport("libc")]`.

**Comment by somdoron** [comment]
we can check that the platform is not windows (or is posix) and the version of mono and only then use the workaround...

**Comment by WimObiwan** [comment]
This issue is finally solved in Mono 4.2.0...
https://bugzilla.xamarin.com/show_bug.cgi?id=30880 
http://www.mono-project.com/docs/about-mono/releases/4.2.0/  ==> #30080 is included
